src/test.js

```diff
@@ -344,13 +344,15 @@ Test.prototype = {
 			// generating stack trace is expensive, so using a getter will help defer this until we need it
 			get source() { return test.stack; }
 		} ).then( function() {
-			if ( module.testsRun === numberOfTests( module ) ) {
+			if ( ( module.testsRun + module.testsNotRan ) === numberOfTests( module ) ) {
 				const completedModules = [ module ];
 
 				// Check if the parent modules, iteratively, are done. If that the case,
 				// we emit the `suiteEnd` event and trigger `moduleDone` callback.
 				let parent = module.parentModule;
-				while ( parent && parent.testsRun === numberOfTests( parent ) ) {
+				while (	parent &&
+					( parent.testsRun + parent.testsNotRan ) === numberOfTests( parent )
+				) {
```

Comment:
Do you think it's odd to signal that this module is completed when it really hasn't? I understand the reasoning, but it feels like you may be giving folks a false sense of security. I also understand the intent is to notify that some tests weren't run. Do we think this will mitigate the concern here?

Reply:
@scalvert this is a good question. 
It depends on what is the intent of `moduleDone`:
1) done executing all the tests listed in the module.
2) done executing all tests picked to be ran in the module.

I implemented with the latter understanding. 

From the qunit documentation:
https://api.qunitjs.com/callbacks/QUnit.moduleDone
`Register a callback to fire whenever a module ends. The callback can return a promise that will be waited for before the next callback is handled.`

Reply:
I agree with @step2yeung's interpretation here, IMHO `moduleDone` should be called when the module is finished running tests (regardless if all tests were ran, some were skipped, some were filtered, etc). This should conceptually be not that different from the `hooks.after` callback timing.

Reply:
My last comment made me wonder if we should be creating another mechanism for tracking the same thing. Why can't we use the same technique that we use to detect when to run `after` hooks for the module? Does it have the same issue (if yes, we should include that fix here...)?

Reply:
@rwjblue Good call on the `after` hook, i found that it suffers from the same issue. Added this fix too now!